Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/AppCenterContentPage.xaml.cs

```diff
@@ -42,6 +46,26 @@ protected override async void OnAppearing()
                 Application.Current.Properties[Constants.UserId] = text;
                 await Application.Current.SavePropertiesAsync();
             };
+
+            StorageMaxSize.Completed += async (sender, args) =>
+            {
+                HandleStorageMaxSizeChanged();
+                await Application.Current.SavePropertiesAsync();
+            };
+
+            StorageMaxSize.Unfocused += async (sender, args) => 
+            {
+                HandleStorageMaxSizeChanged();
+                await Application.Current.SavePropertiesAsync();
+            };
+        }
+
+        private void HandleStorageMaxSizeChanged()
+        {
+            var inputText = StorageMaxSize.Text;
+            var size = string.IsNullOrEmpty(inputText) ? 0 : long.Parse(inputText);
+            AppCenter.SetMaxStorageSizeAsync(size).Wait();
```

Comment:
Make the function async instead of `Wait`

Reply:
Fixed